tools/src/main/java/io/github/ygojson/tools/dataprovider/domain/uuid/YgojsonIDGenerator.java

```diff
@@ -0,0 +1,145 @@
+package io.github.ygojson.tools.dataprovider.domain.uuid;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.github.f4b6a3.uuid.alt.GUID;
+
+import io.github.ygojson.model.data.Card;
+import io.github.ygojson.model.data.Print;
+import io.github.ygojson.model.data.Set;
+import io.github.ygojson.model.data.definition.Identifiers;
+
+/**
+ * Generator for IDs for YGOJSON.
+ */
+public class YgojsonIDGenerator {
+
+	/**
+	 * Container for the result of the generation of the ID.
+	 * <br>
+	 * This is required to inform the caller if the generated ID is random
+	 * or based on data.
+	 *
+	 * @param id the generated ID
+	 * @param isRandom if the ID is randomly generated instead of using the defined IDs.
+	 */
+	public record YgojsonID(UUID id, boolean isRandom) {}
```

Comment:
Instead of this I would say that we should not generate an UUID and instead return the NIL one. How to handle objects without enough data to generate an UUID is not the responsibility of the generator.